Applications/LLaMA/PyTorch/weights_converter.py

```diff
@@ -1,58 +1,63 @@
-# SPDX-License-Identifier: Apache-2.0
-# Copyright (C) 2023 Seungbaek Hong <sb92.hong@samsung.com>
-# 
-# @file weights_converter.py
-# @date 13 October 2023
-#
-# @author Seungbaek Hong <sb92.hong@samsung.com>
+"""
+SPDX-License-Identifier: Apache-2.0
+Copyright (C) 2023 Seungbaek Hong <sb92.hong@samsung.com>
+
+@file weights_converter.py
+@date 13 October 2023
+
+@author Seungbaek Hong <sb92.hong@samsung.com>
+"""
 
 import torch
 import numpy as np
 from transformers import LlamaForCausalLM
 
-##
-# @brief convert and save weights as nntrainer format for multi-head attention model
-def save_llama_for_nntrainer(params, n_layers, file, dtype):        
+
+def save_llama_for_nntrainer(params, n_layers, file, dtype):
+    """
+    @brief convert and save weights as nntrainer format for multi-head attention model
+    """
+
     def save_weight(weight):
         np.array(weight, dtype=dtype).tofile(file)
 
     def save_embedding(weight):
         save_weight(weight)
 
-    def save_attention(weights, layer_name):        
-        save_weight(params[layer_name + 'input_layernorm' + '.weight'])
-        save_weight(params[layer_name + 'self_attn.q_proj' + '.weight'].permute(1, 0)) 
-        save_weight(params[layer_name + 'self_attn.k_proj' + '.weight'].permute(1, 0))
-        save_weight(params[layer_name + 'self_attn.v_proj' + '.weight'].permute(1, 0))
-        save_weight(params[layer_name + 'self_attn.o_proj' + '.weight'].permute(1, 0))
+    def save_attention(layer_name):
+        save_weight(params[layer_name + "input_layernorm" + ".weight"])
+        save_weight(params[layer_name + "self_attn.q_proj" + ".weight"].permute(1, 0))
+        save_weight(params[layer_name + "self_attn.k_proj" + ".weight"].permute(1, 0))
+        save_weight(params[layer_name + "self_attn.v_proj" + ".weight"].permute(1, 0))
+        save_weight(params[layer_name + "self_attn.o_proj" + ".weight"].permute(1, 0))
 
-    def save_feed_forward(weights, layer_name):
-        save_weight(params[layer_name + 'post_attention_layernorm' + '.weight'])
-        save_weight(params[layer_name + 'mlp.up_proj' + '.weight'].permute(1, 0))
-        save_weight(params[layer_name + 'mlp.gate_proj' + '.weight'].permute(1, 0))        
-        save_weight(params[layer_name + 'mlp.down_proj' + '.weight'].permute(1, 0))
+    def save_feed_forward(layer_name):
+        save_weight(params[layer_name + "post_attention_layernorm" + ".weight"])
+        save_weight(params[layer_name + "mlp.up_proj" + ".weight"].permute(1, 0))
```

Comment:
Should we change this?

Reply:
Yes. This part is not essential for functionality, but it is a string representation method according to the PEP8 standard recommended by Python. Although it is a convention, I changed it because I thought it would be better to adhere to the latest standard. (I did not change it manually, but used Black to format it.)

Reply:
PEP8 doe snot enforce double quoted for string.

>String Quotes
In Python, single-quoted strings and double-quoted strings are the same. This PEP does not make a recommendation for this. Pick a rule and stick to it. When a string contains single or double quote characters, however, use the other one to avoid backslashes in the string. It improves readability.